### PR TITLE
Move NServiceBus.Gateway to use Core v8

### PR DESCRIPTION
**Pull request by mauroservienti**
Related to https://github.com/Particular/NServiceBus/pull/5719

### CODE REVIEW

**Comment by mauroservienti** [comment]
This is blocked by NServiceBus.Callbacks, and NServiceBus.Callbacks depends on NServiceBus.Testing

File: src/NServiceBus.Gateway/Config/ChannelCollection.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#if NET452
+#if NET472
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Config/ChannelCollection.cs:1]: isn't this XML based configuration that is supposed to be removed with this major version?

File: src/NServiceBus.Gateway/Gateway.cs [hunk 2]
```diff
@@ -1,275 +0,0 @@
...
-    using Transport;
-
-    /// <summary>
-    /// Used to configure the gateway.
-    /// </summary>
-    [ObsoleteEx(
-        RemoveInVersion = "4.0",
-        TreatAsErrorFromVersion = "3.0",
-        Message = "Use `EndpointConfiguration.Gateway() to enable the gateway.")]
-    public class Gateway : Feature
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Gateway.cs:34]: I don't think this is supposed to be removed completely, the feature should only be made private

File: src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs [hunk 3]
```diff
@@ -27,9 +29,7 @@ public static GatewaySettings Gateway(this EndpointConfiguration config, Gateway
             Guard.AgainstNull(nameof(config), config);
             Guard.AgainstNull(nameof(storageConfiguration), storageConfiguration);
 
-            config.EnableFeature<Features.Gateway>();
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs:30]: this should remain I think

File: src/NServiceBus.Gateway/Settings/GatewaySettings.cs [hunk 4]
```diff
@@ -147,7 +147,7 @@ public void TransactionTimeout(TimeSpan timeout)
 #if NETSTANDARD2_0
             return null;
 #endif
-#if NET452
+#if NET472
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Settings/GatewaySettings.cs:150]: that stuff can probably also be removed given we no longer support the XML config approach?

File: src/NServiceBus.Gateway/Gateway.cs [hunk 5]
```diff
@@ -85,9 +82,9 @@ protected override void Setup(FeatureConfigurationContext context)
                 channelManager,
                 channelReceiverFactory,
                 GetEndpointRouter(context),
-                b.Build<IDispatchMessages>(),
+                b.GetRequiredService<IDispatchMessages>(),
                 storageConfiguration.CreateStorage(b),
-                b.BuildAll<IDataBus>()?.FirstOrDefault(),
+                b.GetServices<IDataBus>()?.FirstOrDefault(),
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Gateway.cs:87]: I'm not sure `GetServices` can ever return `null` but we'd have to check the specs I guess.

File: src/NServiceBus.Gateway.AcceptanceTests/When_a_persistence_does_not_support_gateway_dedup.cs [hunk 6]
```diff
@@ -1,35 +0,0 @@
-﻿namespace NServiceBus.Gateway.AcceptanceTests
-{
-    using AcceptanceTesting;
-    using NUnit.Framework;
-
-    public class When_a_persistence_does_not_support_gateway_dedup : NServiceBusAcceptanceTest
-    {
-        [Test]
-        public void Should_throw_exception()
```
[the remark this answers is not in the document]
Comment (mauroservienti) [reply src/NServiceBus.Gateway.AcceptanceTests/When_a_persistence_does_not_support_gateway_dedup.cs:9]: Correct

**Pull request merged**